rust/arrow/src/array/array.rs

```diff
@@ -2592,6 +2619,15 @@ mod tests {
         assert_eq!(DataType::Int32, list_array.value_type());
         assert_eq!(3, list_array.len());
         assert_eq!(0, list_array.null_count());
+        assert_eq!(
+            3,
+            list_array
+                .value(0)
+                .as_any()
+                .downcast_ref::<Int32Array>()
+                .unwrap()
+                .value(0)
+        );
```

Comment:
The rest of the added tests would have been passing previously. This would be the failing test. The actual value that was being returned was 0.